### PR TITLE
feat(mech loadout): editable system order

### LINKED ISSUE #1828: [FEATURE] Reorder Systems on a Mech
**Is your feature request related to a problem? Please describe.**
For a system-focused player the order of the systems on a mech can be important to taking one's turn's quickly. However, currently systems can not be reordered, they must be removed and added in the correct order.

**Describe the solution you'd like**
**Being able to reorder systems without removing them.** Click and holding to drag them up and down would be nice and clean, but other methods would be fine.

**Describe alternatives you've considered**
Removing and adding the systems in the correct order is the current solution, but it is frustrating, requires memory, and is not easily changeable.

**Additional context**
I understand this is a QOL improvement, so it is probably pretty low-priority, but an update addressing multiple QOL issues would go a long way in improving the UX for comp/con.

### PR DESCRIPTION
**Pull request by super-salad**
# Description

This is a draft PR to add an editable system order to a mech loadout. Currently the order is only saved when logged in, but that is because of issue #1989. 

Besides wondering if this is a good approach (adding a sort index on to the base IEquipmentData seems useful, but maybe we don't want to be adding a lot of additional properties there?), there is something I need some additional eyes on: Adding the draggable tag on the Active Loadout system block breaks the two-column layout there and forces it into a single column. From looking at the vuedraggable docs, this seems like what we should expect without make some other structural changes to the page layout, but I'm not familiar enough with vue or vuedraggable to know what that would be or if I'm just missing something simple.

The order set from the non-active Mech Loadout is respected on active, however, so perhaps if it's not trivial to make the ActiveSystemsBlock play nicely we leave it at that?

## Issue Number
Closes #1828

## Type of change
- [ x ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by msprijatelj** [comment]
I would need to take a deeper look at this, but I believe a Sort Index may be unnecessary? Ideally we’d just overwrite the mech systems with the newly-ordered systems and save it.

File: src/ui/components/panels/loadout/active_loadout/_ActiveSystemsBlock.vue [hunk 1]
```diff
@@ -68,6 +76,18 @@ export default Vue.extend({
     moddedWeapons() {
       return this.mech.MechLoadoutController.ActiveLoadout.Weapons.filter(x => x.Mod)
     },
+    systems(){
+      return this.mech.MechLoadoutController.ActiveLoadout.Systems
+    },
```
Comment (msprijatelj) [inline src/ui/components/panels/loadout/active_loadout/_ActiveSystemsBlock.vue:81]: We would want to define “systems” under the component’s “data”, initializing it with a “mounted” or “created” block. Then whenever we experience a “move” event, we save the component’s “systems” to the mech’s “systems”.

**Review by msprijatelj — COMMENTED** [review]
Yeah, I think you’ll definitely want to tear out the Sort index stuff. VueDraggable will let us manage a list’s order without needing to manually set an index.

**Comment by super-salad** [comment]
Ok, I know it's been a long time (past few months have been crazy!) but I get get around to implementing your feedback @msprijatelj.

I removed the draggable changes from the _ActiveSystemsBlock because for the life of me I can't figure out how to preserve the two column layout. My vue fu needs to be stronger. The good thing is that the custom order is still respected on that page, just not editable directly. Does this look like the right direction here?

**Comment by msprijatelj** [comment]
I’ll take a look at this when I can, but I will say that preventing reordering of systems in Active Mode seems reasonable at the moment; during active mode I would expect less rearranging and more playing, anyway.

**Review by jarena3 — APPROVED** [review]

**Pull request merged**